contracts/Core/StakeManager.sol

```diff
@@ -183,13 +183,18 @@ contract StakeManager is Initializable, StakeStorage, StateManager, Pause, Stake
         require(locks[msg.sender][staker.tokenAddress].amount == 0, "Existing Lock");
         require(sAmount > 0, "Non-Positive Amount");
 
-        // slither-disable-next-line reentrancy-events,reentrancy-no-eth
-        rewardManager.giveInactivityPenalties(epoch, stakerId);
-
         IStakedToken sToken = IStakedToken(staker.tokenAddress);
         require(sToken.balanceOf(msg.sender) >= sAmount, "Invalid Amount");
+        uint256 totalSupply = sToken.totalSupply();
+        uint256 rAmount = _convertSRZRToRZR(sAmount, staker.stake, totalSupply);
+        uint256 prevStake = staker.stake;
+
+        if ((staker.stake - rAmount) < minStake) {
+            // slither-disable-next-line reentrancy-events,reentrancy-no-eth
+            rewardManager.giveInactivityPenalties(epoch, stakerId);
```

Comment:
we are supposed to do this for randao penalty and not for inactive penalty.
randao penalty is small but inactive penalty can be very big

if staker is inactive for thousands of epochs and inactivity penalty > min stake, staker can escape inactivity penalty by doing this (he will lose minstake instead)

Reply:
Will have to separate out randao and inactivity into 2 different functions to implement this. Randao is anyway getting removed in #649 so this can be considered as a quick fix.

What say? @abhishekvispute @SkandaBhat @hrishikeshio

Reply:
hey 
cant we do this check on this line on the top of randao penalty only 
https://github.com/razor-network/contracts/blob/580af180bf20d6ca3ab487428725f15adaf9afae/contracts/Core/RewardManager.sol#L88

so inactivity will be as it is 
and for randao we will have this check

Reply:
hmm but then this will be applicable for commit as well nvm

Reply:
Ya then now we have two options 
1) Divide the function for only randao
2) Use above and let staker escape inactivity if > minstake
3) Disallow unstake in commit and reveal

Now if its just for quickfix : 
I think 3) is better option for a while, its not creating any new issue and a ux concern.

Reply:
Ya makes sense. Will do this